modal_utils/grpc_testing.py

```diff
@@ -0,0 +1,125 @@
+import contextlib
```

Comment:
This file was taken verbatim from our internal repo, and changed to use `typing.List` and `typing.Tuple` instead of builtin generics `list` and `tuple`